pkg/providers/openstack/run_instance.go

```diff
@@ -1,10 +1,90 @@
 package openstack
 
 import (
+	"github.com/Sirupsen/logrus"
 	"github.com/emc-advanced-dev/pkg/errors"
 	"github.com/emc-advanced-dev/unik/pkg/types"
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack/compute/v2/servers"
+	"time"
 )
 
+const DEFAULT_INSTANCE_DISKMB int = 10 * 1024 // 10 GB
+
 func (p *OpenstackProvider) RunInstance(params types.RunInstanceParams) (_ *types.Instance, err error) {
-	return nil, errors.New("not yet supportded for openstack", nil)
+	// return nil, errors.New("not yet supportded for openstack", nil)
+
+	logrus.WithFields(logrus.Fields{
+		"image-id": params.ImageId,
+		"mounts":   params.MntPointsToVolumeIds,
+		"env":      params.Env,
+	}).Infof("running instance %s", params.Name)
+
+	clientNova, err := p.newClientNova()
+	if err != nil {
+		return nil, err
+	}
+
+	image, err := p.GetImage(params.ImageId)
+	if err != nil {
+		return nil, errors.New("failed to get image", err)
+	}
+
+	// If not set, use default.
+	if params.InstanceMemory <= 0 {
+		params.InstanceMemory = image.RunSpec.DefaultInstanceMemory
+	}
+
+	// Pick flavor.
+	minDiskMB := image.RunSpec.MinInstanceDiskMB
+	if minDiskMB <= 0 {
+		minDiskMB = DEFAULT_INSTANCE_DISKMB // TODO: raise error here, since compiler should set MinInstanceDiskMB
```

Comment:
Was this TODO meant to be resolved in this commit? If not, please make it more explicit, as well as add `TODO(miha-plesko)` so that it is more clear who is to resolve it.